src/Access/Dialog.vala

```diff
@@ -54,60 +54,56 @@ public class Access.Dialog : Granite.MessageDialog {
     }
 
     construct {
-        skip_taskbar_hint = true;
+        unowned var surface = get_surface ();
+        if (surface is Gdk.X11.Surface) {
+            unowned var x11_surface = (Gdk.X11.Surface) surface;
+            x11_surface.set_skip_taskbar_hint (true);
+        }
+
         resizable = false;
         modal = true;
 
         choices = new List<Choice> ();
-        set_role ("AccessDialog"); // used in Gala.CloseDialog
-        set_keep_above (true);
 
         if (app_id != "") {
             badge_icon = new DesktopAppInfo (app_id + ".desktop").get_icon ();
         }
 
         deny_button = add_button (_("Deny Access"), Gtk.ResponseType.CANCEL) as Gtk.Button;
         grant_button = add_button (_("Grant Access"), Gtk.ResponseType.OK) as Gtk.Button;
-        unowned var grant_context = grant_button.get_style_context ();
 
         if (action == ButtonAction.SUGGESTED) {
-            grant_context.add_class (Gtk.STYLE_CLASS_SUGGESTED_ACTION);
-            set_default (grant_button);
+            grant_button.add_css_class ("suggested-action");
+            set_default_widget (grant_button);
         } else {
-            grant_context.add_class (Gtk.STYLE_CLASS_DESTRUCTIVE_ACTION);
-            set_default (deny_button);
+            grant_button.add_css_class ("destructive-action");
+            set_default_widget (deny_button);
         }
 
         box = new Gtk.Box (Gtk.Orientation.VERTICAL, 6);
-        custom_bin.child = box;
+        custom_bin.append (box);
         box.show ();
 
-        if (parent_window != "") {
-            realize.connect (() => {
+        if (parent_window == "") {
+            warning ("Unknown parent window: The portal dialog might appear at the wrong spot because we can't assign a transient parent.");
+
+        } else {
+            ((Gtk.Widget) this).realize.connect (() => {
```

Comment:
An empty string is an vaild argument, we shouldn't warning it that case, only when we have the parent and can't set the dialog transient for it.

Also, the `if (parent_window != "")` check need to be done inside the realize now that we also need to call `set_skip_taskbar_hint` there.
```suggestion
        realize.connect (() => {
            var surface = get_surface ();

            if (surface is Gdk.X11.Surface) {
                ((Gdk.X11.Surface) surface).set_skip_taskbar_hint (true);
            }
            
            if (parent_window != "") {
```